Crawler/LexicalAnalyzer/TokenDefinition.cs

```diff
@@ -4,20 +4,21 @@ namespace Crawler.LexicalAnalyzer
 {
     public class TokenDefinition
     {
-        private readonly Regex regex;
-        private readonly eTokenType tokenType;
+        private Regex regex = new Regex(string.Empty);
 
-        public TokenDefinition(eTokenType tokenType, string regexPattern)
+        public eTokenType TokenType { get; set; }
+
+        public string Pattern
         {
-            regex = new Regex(regexPattern);
-            this.tokenType = tokenType;
+            get => regex.ToString();
+            set => regex = new Regex(value);
```

Comment:
Previously to these changes "regex" was read only, but currently it can be changed by the "Pattern" property. Has this been intentional?

Reply:
Yes because it loads from the config file the .net core framework uses the properties to set the class. the regex member can not be readonly because it is set by a property and not only by ctor